microsoft/vs-solutionpersistence
# Round trip legacy VS properties in slnx, and add support to trim when…

- Round trip legacy VS properties in slnx (e.g. Project/Solution Id)
- And add support to trim VS properties when serializing (on by default when converting sln to slnx)
- Move VS specific properties from solution to "Visual Studio" property bag. Added helper to get/set known values.
- Don't save 'OpenWith' Visual Studio property since we don't have support for it yet.
- Change default CLR project types from CPS type ids to standard type ids, don't ever save CPS project ids to slnx.
- Add file versioning to slnx and fail if loading future major version in case we need it in the future.
- Small Fixes
  - Attempt to resolve project type name if user sets project type id in model.
  - Fix slnx not reading property bag scope value.
  - Add copy constructor to serializer settings to make it easier to change a setting.
- Tests
  - Add test for configurations, valid names, vs properties in slnx, file versioning,
  - Move keyword string debug check code to test. 
  - Remove most usages of backslash in test code to reduce likelihood of breaking Linux tests.
  - Add comments to all tests and add warnings for missing comments.
